src/smt.rs

```diff
@@ -789,18 +848,25 @@ impl SparseMerkleTree {
         })
     }
 
-    pub fn verify(
-        query_keys: &Vec<Vec<u8>>,
-        proof: &Proof,
-        root: &[u8],
-        key_length: usize,
-    ) -> Result<bool, SMTError> {
-        if query_keys.len() != proof.queries.len() {
-            return Ok(false);
+    fn filter_map(proof: &Proof) -> HashMap<Vec<bool>, QueryProofWithProof> {
```

Comment:
We discuss and agreed that inside `struct` implementation all `fn` should be implemented before all `pub fn`.
Please check also all other `structs` which are related to this PR.

Reply:
Done.